File: Práctica 2/Localidad.cc
//DNI 23900715N HERNANDEZ GOMEZ, JESSICA
#include "Localidad.h"

ostream & operator<< (ostream &os, const Localidad &l) { //Funcion amiga de sobrecarga del operador salida
    os << l.id << "-" << l.nombre << "-" << l.coor << endl << l.info;
    return os;
}

Localidad::Localidad(){
    nombre = "";
    coor = Coordenadas();
    info = InfoTur();
    id = -1;
}

Localidad::Localidad(string n){
    nombre = n;
    coor = Coordenadas();
    info = InfoTur();
    id = -1;
}

Localidad::Localidad(const Localidad &l){
    nombre = l.nombre;
    coor = l.coor;
    info = l.info;
    id = l.id;
}

Localidad::~Localidad(){
    nombre = "";
    coor = Coordenadas();
    info = InfoTur();
    id = -1;
}

Localidad & Localidad::operator=(const Localidad &l){
    nombre = l.nombre;
    coor = l.coor;
    info = l.info;
    id = l.id;

    return *this;
}

int Localidad::setCoor(int f, int c, vector<vector<char> > &mapa){
    int devolver = -1;

    if(f >= 0 && f<(int)mapa.size()){
        if(c >= 0 && c < (int)mapa[f].size()){
            if(id == -1){
                if(mapa[f][c] == 'T'){
                    coor.setFila(f);
                    coor.setColumna(c);
                    mapa[f][c] = 'L';
                    id = mapa[1].size() * f + c;
                    devolver = id;
                }
            }
        }
    }
    return devolver;
}

void Localidad::setInfo(InfoTur i){
    info = i;
}

string Localidad::getNombre(){
    return nombre;
}

Coordenadas Localidad::getCoor(){
    return coor;
}

InfoTur &Localidad::getInfo(){
    return info;
}

int Localidad::getId(){
    return id;
}

File: Práctica 1/Java/Localidad.java
//23900715N HERNANDEZ GOMEZ, JESSICA
import java.util.*;

public class Localidad {
	private String nombre;
	private Coordenadas coor;
	private InfoTur info;
	private int id;
	
	public Localidad(String n){
		nombre = n;
		coor = new Coordenadas();
		info = new InfoTur();
		id = -1;
	}
	
	public int setCoor(int f, int c, ArrayList<ArrayList<Character>> mapa){
		int devolver = -1;
		
		if(mapa!=null){
			if (mapa.get(f).get(c)!=null){
				if(mapa.get(f).get(c)!= -1){
					if(mapa.get(f).get(c) == 'T'){
						coor.setFila(f);
						coor.setColumna(c);
						mapa.get(f).set(c, 'L');
						id = mapa.get(1).size() * f + c;
						devolver = id;
					}
				}
			}
		}
		
		return devolver;
	}
	
	public void setInfo(InfoTur it){
		info = it;
	}

	public String getNombre() {
		return nombre;
	}

	public Coordenadas getCoor() {
		return coor;
	}

	public InfoTur getInfo() {
		return info;
	}

	public int getId() {
		return id;
	}
	
	public String toString(){
		String devolver = id + "-" + nombre + "-" + coor.toString() + info.toString();
		return devolver;
	}
}
